src/store/diagnostics.ts

```diff
@@ -0,0 +1,112 @@
+import { ActionContext } from './ts-helpers';
+
+import type { ServerState } from '@/main/commandServer/httpCommandServer';
+import { DiagnosticsCheck } from '@/main/diagnostics/diagnostics';
+
+interface DiagnosticsState {
+  diagnostics: Array<DiagnosticsCheck>,
+  timeLastRun: Date;
+  inError: boolean;
+}
+
+const uri = (port: number, pathRemainder: string) => `http://localhost:${ port }/v0/${ pathRemainder }`;
+
+export const state: () => DiagnosticsState = () => (
+  {
+    diagnostics: [],
+    timeLastRun: new Date(),
+    inError:     false,
+  }
+);
+
+export const mutations = {
+  SET_DIAGNOSTICS(state: DiagnosticsState, diagnostics: DiagnosticsCheck[]) {
+    state.diagnostics = diagnostics;
+    state.inError = false;
+  },
+  SET_TIME_LAST_RUN(state: DiagnosticsState, currentDate: Date) {
+    state.timeLastRun = currentDate;
+  },
+  SET_IN_ERROR(state: DiagnosticsState, status: boolean) {
+    state.inError = status;
+  },
+};
+
+type DiagActionContext = ActionContext<DiagnosticsState>;
+
+export const actions = {
+  async fetchDiagnostics({ state, commit }: DiagActionContext, args: ServerState) {
+    const rows: Array<DiagnosticsCheck> = [];
+    const {
+      port,
+      user,
+      password,
+    } = args;
+    const response = await fetch(
+      uri(port, 'diagnostic_categories'),
+      {
+        headers: new Headers({
+          Authorization:  `Basic ${ window.btoa(`${ user }:${ password }`) }`,
+          'Content-Type': 'application/x-www-form-urlencoded',
+        }),
+      });
+
+    if (!response.ok) {
+      commit('SET_IN_ERROR', true);
+
+      return;
+    }
+
+    const categories: string[] = await response.json();
+
+    for (const category of categories) {
+      const response = await fetch(
+        uri(port, `diagnostic_ids?category=${ category }`),
+        {
+          headers: new Headers({
+            Authorization:  `Basic ${ window.btoa(`${ user }:${ password }`) }`,
+            'Content-Type': 'application/x-www-form-urlencoded',
+          }),
+        });
+
+      if (!response.ok) {
+        commit('SET_IN_ERROR', true);
+
+        return;
+      }
+      const checkIDs: string[] = await response.json();
+
+      for (const checkID of checkIDs) {
+        const response = await fetch(
+          uri(port, `diagnostic_check?category=${ category }&id=${ checkID }`),
+          {
+            headers: new Headers({
+              Authorization:  `Basic ${ window.btoa(`${ user }:${ password }`) }`,
+              'Content-Type': 'application/x-www-form-urlencoded',
+            }),
+          });
+
+        if (!response.ok) {
+          commit('SET_IN_ERROR', true);
+
+          return;
+        }
+        const res = await response.json();
+
+        res.category = category;
+        rows.push(res);
+      }
+    }
```

Comment:
This seems a little chatty to me because we're nesting two loops to fetch each diagnostic check. Would it make sense to fetch all of our diagnostics in a single request?

Reply:
I changed `/v0/diagnostic_check` to `/v0/diagnostic_checks` and simplified its use so by default it returns all the checks.

Currently all checks are assumed to have failed, but eventually each check will have a status field, and we will only ever return checks where the status field indicates failure.